src/Http/Controllers/VatController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Rapidez\VatValidation\Http\Controllers;
+
+use Ibericode\Vat\Validator;
+use Ibericode\Vat\Vies\ViesException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+
+class VatController
+{
+    /** @return array<string, mixed> */
+    public function __invoke(Request $request): array
+    {
+        $request->validate([
+            'id' => 'string|required',
+        ]);
+
+        if (!config('rapidez.vat-validation.allow-anywhere')) {
+            // Check if we should validate here
+            abort_unless($request->bearerToken() || $request->headers->get('referer') === route('account.register'), 401);
+
+            if ($request->bearerToken()) {
+                config('rapidez.models.quote')::whereQuoteIdOrCustomerToken($request->bearerToken())->firstOrFail();
+            }
+        }
+
+        try {
+            // Try validating the number
+            $validator = new Validator;
+            $result = Cache::remember('vat_' . $request->id, config('rapidez.vat-validation.cache-duration'), function () use ($request, $validator) {
+                return $validator->validateVatNumber($request->id);
+            });
+
+            return ['result' => $result];
```

Comment:
Should we also return a 4xx response on an incorrect vat? 
https://github.com/rapidez/vat-validation/pull/1/files/33022e7a5301702f72834832d9aa22d6304ce132#r1796635804

Reply:
Personally returning a 4xx when the VAT is invalid seems improper, given that something being invalid is still a valid output of a check.

I would expect a `true` or `false` output, not a `true` or `ERROR 4xx` output.

Reply:
You could still return false on an error 4xx.
As outlined in https://github.com/rapidez/vat-validation/pull/1#discussion_r1796635804
this is inconsistent with Laravel's validator method.

We could've created a [custom validation rule](https://laravel.com/docs/11.x/validation#custom-validation-rules) for a valid VAT, resulting in Laravel returning a 4xx response.
Same as how it will return a 4xx now if no vat id is provided, but a 2xx if anything is provided in the vat id field

Reply:
Sounds fun until you end up getting [a FetchError just because the status code is a 4xx](https://github.com/rapidez/core/blob/master/resources/js/fetch.js#L60) 🙂 we would have to work around these pre-existing error handling routines just to get the same effect as just returning `true` or `false`, and I don't think I prefer that.

Again, I also don't think that returning a client or server error is the right call in this case. You also don't get an error out of the `isEmailAvailable` magento rest api query when an email is not available, for example.

Reply:
While it would happen more often we would be getting that without returning a 4xx ourselves as well.
Laravel returns a 4xx on validation failure too so we must account for that anyways.

I don't think we need to work around pre-existing error handling, we simply need to _handle_ the error.
Luckily that's an easy fix already implemented here for Graphql https://github.com/rapidez/core/blob/a93fbc0de64a55559149ad0665a8d05339682c17/resources/js/components/GraphqlMutation.vue#L125

So during our catch we just need to check some additional things and change the return value:
```js
return await window
    .rapidezAPI('post', 'vat-validate', data, options)
        .catch(async (error) => {
            if (FetchError.prototype.isPrototypeOf(error)) {
                // Now we know for sure error.response is available
                if (error.response.status === 422) {
                    // Validation did not pass since laravel returned a 422 https://laravel.com/docs/11.x/validation#quick-xhr-requests-and-validation
                    return false
                }
             }

            window.Notify(window.config.translations.errors.wrong, 'error')
       	    return 'error'
        })
```

Of course we can shorten it to a single if statement if we like

Reply:
I've made an internal issue for this specific thing for now (key: RAP-948) 🙂